FinniversKit/Sources/SwiftUI Components/Button/SwiftUIFloatingButton.swift

```diff
@@ -0,0 +1,92 @@
+import SwiftUI
+
+public struct SwiftUIFloatingButton: View {
+    @ObservedObject var viewModel: SwiftUIFloatingButtonModel
+    @State private var isPressed: Bool = false
+    private let style: SwiftUIFloatingButtonStyle
+    private let action: () -> Void
+
+    public init(
+        style: SwiftUIFloatingButtonStyle = .default,
+        viewModel: SwiftUIFloatingButtonModel,
+        action: @escaping () -> Void
+    ) {
+        self.style = style
+        self.viewModel = viewModel
+        self.action = action
+    }
+
+    public var body: some View {
+        SwiftUI.Button(action: {
+            action()
+            isPressed = false
+        }, label: {
+            HStack {
+                viewModel.image
+                    .foregroundColor(style.tintColor)
+                if let title = viewModel.title {
+                    Text(title)
+                        .finnFont(.detail)
+                        .foregroundColor(style.titleColor)
+                }
+            }
```

Comment:
Maybe all modifiers underneath can be can be added into a custom button style that can be applied with `.buttonStyle(_)`

Reply:
Ah, of course! I think that pretty much eliminates the need for a custom view altogether:

```swift
struct FloatingButtonStyle: ButtonStyle {
    let backgroundColor: Color
    let backgroundColorPressed: Color
    let shadowColor: Color
    let shadowOffset: CGSize
    let shadowRadius: CGFloat

    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .padding(.spacingM)
            .background(configuration.isPressed ? backgroundColorPressed : backgroundColor)
            .clipShape(Capsule())
            .shadow(
                color: shadowColor,
                radius: shadowRadius,
                x: shadowOffset.width,
                y: shadowOffset.height
            )
    }
}
```